### ISSUE TITLE
Empty string in query

### ISSUE THREAD

**Issue by ghendo** [mongodb-labs/mongo-csharp-search#10]
With search autocomplete if the query is an empty string then crashes
I feel this is not the right thing

**Comment by dgolub** [comment]
The behavior that I'm seeing is not a crash per se but rather an exception indicating that the input is invalid:
```
MongoDB.Driver.MongoCommandException : Command aggregate failed: PlanExecutor error during aggregation :: caused by :: Remote error from mongot :: caused by :: "autocomplete.query" cannot be empty.
```
Is this what you're seeing?  We could probably add some client-side error handling to avoid the request to the server and provide a shorter error message, but Atlas Search doesn't accept empty query strings for autocomplete queries.  Either way, the best course of action is going to be to add a check for empty strings in your application code and handle that case as per your needs.

**Comment by ghendo** [comment]
Yes that is what I am seeing.  I feel that in a common autocomplete senario where a user will start to type into a text box to perform a search that obviously they start with an empty box. I cant see any reason why empty string is some special case that should be an error. In my mind it just should return everything. Anyhow if thats how its works ill have to deal with it.

**Comment by ghendo** [comment]
Further testing showed that empty string will work if MinimumShouldMatch is set > 0

**Comment by dgolub** [comment]
What behavior instead of an error would be useful for your use case?  An empty result set?

**Comment by ghendo** [comment]
Well my feeble mind thought if no filter then return all the things
